### ISSUE TITLE
Reported Energy Usage is Significantly Higher Than Actual Usage

### ISSUE THREAD

**Issue by nuttytree** [nuttytree/ESPHome-Devices#15]
Energy values remain at 0 while the device is off as expected but as soon as it is turned on it immediately jumps to some unrealistic value.

**Comment by nuttytree** [comment]
Digging into this I discovered that is an issue with how the ESPHome Total Daily Energy sensor calculates values.  It is fine for power sensors that update very frequently but for devices that update infrequently the calculation is way off.  I have submitted a PR esphome/esphome#2163 to update the Total Daily Energy sensor to have different options for the calculation similar to the Integration sensor.

**Comment by nuttytree** [comment]
I tried to pull in the Total Daily Energy sensor as an external component directly from the esphome branch that it was updated in but there is a change to the sensor schema that exists in dev but is not part of version 1.20.4.  So I ended up pulling a copy into this repo and modifying it to work with the 1.20.4 version of the sensor schema.  Which this brings up the topic of versioning.  This is not the first time I have either had conflicts with my code and new versions of ESPHome.  So my plan going forward is to start tagging this repo with the version of ESPHome I am currently running it on so it is easier for everyone pulling in my custom components to get the correct version.
